vms/platformvm/state/metadata_delegator.go

```diff
@@ -9,17 +9,35 @@ import (
 )
 
 type delegatorMetadata struct {
-	PotentialReward uint64
+	PotentialReward uint64 `v0:"true"`
+	StakerStartTime int64  `          v1:"true"`
 
 	txID ids.ID
 }
 
 func parseDelegatorMetadata(bytes []byte, metadata *delegatorMetadata) error {
-	var err error
-	metadata.PotentialReward, err = database.ParseUInt64(bytes)
-	return err
+	switch len(bytes) {
+	case database.Uint64Size:
+		// only potential reward was stored
+		var err error
+		metadata.PotentialReward, err = database.ParseUInt64(bytes)
+		if err != nil {
+			return err
+		}
+
+	default:
+		_, err := metadataCodec.Unmarshal(bytes, metadata)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
 func writeDelegatorMetadata(db database.KeyValueWriter, metadata *delegatorMetadata) error {
-	return database.PutUInt64(db, metadata.txID[:], metadata.PotentialReward)
+	metadataBytes, err := metadataCodec.Marshal(v1, metadata)
```

Comment:
we store start time as soon as we deploy code, without waiting for fork activation

Reply:
The versioning should allow smooth handling of upgrades/downgrades
When reading from disk:

1. If metadata have version zero, start time won't be loaded from disk and the default value AddStakerTx.StartTime() will be used.
2. If metadata have version 1, start time will be read from disk.

This is why we don't need a data migration here, which may cause problems with downgrades.
The downside is that it'd be writing the StartTime to disk as soon as the code is deployed, even if Durango is not active. This is an extra int64 written for each staker promoted from pending to current among code deploy and code activation.

Reply:
What happens when I upgrade my node to this PR and write codec version v1 and then downgrade to a node that does not support parsing codec version v1?

Reply:
@dhrubabasu right! I fixed it by reintroducing the config.Confg and by selecting codec version based on the fork activation. Code will be cleaned up as soon as fork is activated